android/src/main/java/com/swmansion/reanimated/NodesManager.java

```diff
@@ -137,6 +140,17 @@ private void onAnimationFrame(long frameTimeNanos) {
       Node.runUpdates(updateContext);
     }
 
+    if (updateContext.shouldTriggerUIUpdate) {
+      mContext.runOnNativeModulesQueueThread(
+              new GuardedRunnable(mContext) {
+                @Override
+                public void runGuarded() {
+                  mUIManager.onBatchComplete();
```

Comment:
I think we shouldn't just call `onBatchComplete` like that. This may intercept batched calls being made at the same time from JS. Would be the best to detect if we are in the middle of a batch and if so we can skip calling `onBatchComplete` and rely on getting it dispatched from JS thread. Also May be worth enqueuing all the UIManager operations here such that we can call it from a single runnable without risking some other operations being enqueued in the meantime